src/extensions/firetrack/FireTrackView/FireTrackView.tsx

```diff
@@ -1,13 +1,17 @@
-import React, { FunctionComponent, useCallback } from 'react'
-import { SortingViewProps } from "../../extensionInterface"
+import { Table, TableBody, TableCell, TableHead, TableRow } from '@material-ui/core'
+import React, { FunctionComponent, useCallback, useEffect, useReducer, useState } from 'react'
+import { SortingSelection, SortingViewProps } from "../../extensionInterface"
+import useTimeseriesData, { TimeseriesData } from '../../timeseries/TimeseriesViewNew/useTimeseriesModel'
 
-const FireTrackView: FunctionComponent<SortingViewProps> = ({recording, sorting, selection, selectionDispatch}) => {
+const FireTrackView: FunctionComponent<SortingViewProps> = ({recording, sorting, selection, selectionDispatch, hither}) => {
     const handleStopAnimation = useCallback(( args) => {
         selectionDispatch({type: 'SetCurrentTimepointVelocity', velocity: 0})
     }, [ selectionDispatch ])
     const handleStartAnimation = useCallback(( args) => {
         selectionDispatch({type: 'SetCurrentTimepointVelocity', velocity: 200})
     }, [ selectionDispatch ])
+
+    const timeseriesData = useTimeseriesData(recording.recordingObject, recording.recordingInfo, hither)
```

Comment:
Simplify the timeseries model system with this custom hook: useTimeseriesData(...)